metagraph/src/graph/representation/hash/dbg_sshash.cpp

```diff
@@ -0,0 +1,494 @@
+#include "dbg_sshash.hpp"
+
+#include <sdsl/uint256_t.hpp>
+
+#include "common/seq_tools/reverse_complement.hpp"
+#include "common/threads/threading.hpp"
+#include "common/logger.hpp"
+#include "kmer/kmer_extractor.hpp"
+
+
+namespace mtg {
+namespace graph {
+
+const std::string DBGSSHash::alphabet_ = kmer::KmerExtractor2Bit().alphabet;
+
+using Kmer64 = uint64_t;
+using Kmer128 = __uint128_t;
+using Kmer256 = sdsl::uint256_t;
+
+template <typename KmerInt>
+class Dictionary : public IDictionary {
```

Comment:
Why do we need this convoluted strcture of Dictionary and IDictionary, rather than using `sshash::dictionary` directly, or at least inhereting from it directly? If it is about `const` qualifiers, would it make sense to add them to `sshash::dictionary` in our `sshash` fork instead?